pkg/hooks/hooks.go

```diff
@@ -0,0 +1,44 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package hooks
+
+import (
+	"encoding/json"
+
+	k8sv1 "k8s.io/api/core/v1"
+)
+
+const HookSidecarListAnnotationName = "hooks.kubevirt.io/hookSidecars"
+const HookSocketsSharedDirectory = "/var/run/kubevirt-hooks"
+
+type HookSidecarList []HookSidecar
+
+type HookSidecar struct {
+	Image           string           `json:"image"`
+	ImagePullPolicy k8sv1.PullPolicy `json:"imagePullPolicy"`
+}
+
+func UnmarshalHookSidecarList(annotation string) (HookSidecarList, error) {
```

Comment:
since you already have a dependency on the k8sv1 api you can expect the k8s.PodSpec.Metadata struct here no ?

Reply:
Good idea